File: api/src/main/java/io/bunnies/baas/resources/types/PosterMediaType.java
package io.bunnies.baas.resources.types;

import io.bunnies.baas.Constants;

public class PosterMediaType extends ResourceMediaType {
    public static final String KEY = "POSTER";

    public PosterMediaType(String resourceID) {
        super(resourceID);
    }

    @Override
    public String constructUrl() {
        return Constants.MEDIA_BASE_URL + "poster/" + this.resourceID + ".png";
    }
}

File: api/src/main/java/io/bunnies/baas/resources/types/PosterMediaType.kt
package io.bunnies.baas.resources.types

class PosterMediaType(resourceID: String, mediaBaseUrl: String) : ResourceMediaType(resourceID, mediaBaseUrl) {

    override fun constructUrl(): String {
        return this.mediaBaseUrl + "poster/" + this.resourceID + ".png"
    }

    companion object {
        val KEY = "POSTER"
    }
}
